builtin/audit/file/backend_test.go

```diff
@@ -0,0 +1,79 @@
+package file
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/vault/audit"
+	"github.com/hashicorp/vault/helper/salt"
+)
+
+func TestAuditFile_fileModeNew(t *testing.T) {
+	salter, _ := salt.NewSalt(nil, nil)
+
+	modeStr := "0777"
+	mode, err := strconv.ParseUint(modeStr, 8, 32)
+
+	path, err := ioutil.TempDir("", "test")
+	defer os.RemoveAll(path)
+
+	file := filepath.Join(path, "auditTest.txt")
+
+	config := map[string]string{
+		"path": file,
+		"mode": modeStr,
+	}
+
+	_, err = Factory(&audit.BackendConfig{
+		Salt:   salter,
+		Config: config,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(file)
+
+	info, _ := os.Stat(file)
+	createdMode := info.Mode()
+	if createdMode != os.FileMode(mode) {
+		t.Fatalf("File mode does not match.")
+	}
+}
+
+func TestAuditFile_fileModeExisting(t *testing.T) {
+	salter, _ := salt.NewSalt(nil, nil)
+
+	f, err := ioutil.TempFile("", "test")
+	if err != nil {
+		t.Fatalf("Failure to create test file.")
+	}
+	defer os.Remove(f.Name())
+	err = f.Close()
+	if err != nil {
+		t.Fatalf("Failure to close the file.")
+	}
+
```

Comment:
By this point you have a file that you've closed but you haven't actually found out the mode, or set it yourself. If the point of this function is to ensure that an existing file is correctly `chmod`ed inside the backend creation function, and you're not checking the actual mode of the file (or setting it yourself), you don't know whether the code in the function is actually doing anything.

You should `chmod` the file to something ahead of time, then verify at the end that it was correctly changed by the backend to the expected value (since no mode value is being passed in).